deepchecks/vision/utils/detection_encoders.py

```diff
@@ -0,0 +1,172 @@
+# Copyright (C) 2021 Deepchecks (https://www.deepchecks.com)
+#
+# This file is part of Deepchecks.
+# Deepchecks is distributed under the terms of the GNU Affero General
+# Public License (version 3 or later).
+# You should have received a copy of the GNU Affero General Public License
+# along with Deepchecks.  If not, see <http://www.gnu.org/licenses/>.
+# ----------------------------------------------------------------------------
+#
+"""Module for defining detection encoders."""
+from collections import Counter
+from typing import Union, Callable, Optional
+
+__all__ = ['DetectionLabelEncoder', 'DetectionPredictionEncoder']
+
+import numpy as np
+import torch
+from torch.utils.data import DataLoader
+
+from .base_encoders import BaseLabelEncoder, BasePredictionEncoder
+from ...core.errors import DeepchecksValueError
+
+
+class DetectionLabelEncoder(BaseLabelEncoder):
+    """
+    Class for encoding the detection annotations to the required format.
+
+    Parameters
+    ----------
+    label_encoder : Union[str, Callable]
+        Function or string that specifies the encoding function.
+        If a string, it must be one of the following:
+            - 'xyxy' - x1, y1, x2, y2 represent the upper left and lower right corners of the bounding box.
+            - 'xywh' - x1, y1, w, h represent the upper left corner of the bounding box and its width and height.
+            - 'cxcywh' - x, y, w, h represent the center of the bounding box and its width and height.
+            - 'xyxyn' - x1, y1, x2, y2, n represent the upper left and lower right corners of the bounding box,
+                        normalized to the image dimensions.
+            - 'xywhn' - x1, y1, w, h, represent the upper left corner of the bounding box and its width and height,
+                        normalized by the image dimensions.
+            - 'cxcywhn' - (YOLO format) x, y, w, h, represent the center of the bounding box and its width and height,
+                          normalized by the image dimensions.
+        In addition, the label shape should be a list of length N containing tensors of shape (M, 5), where N is the
+        number of samples, M is the number of bounding boxes, and each bounding box is represented by 5 values:
+        (class_id, 4 coordinates in the format specified by the `label_encoder` parameter).
+
+        If a function, it must follow the signature:
+        Function that takes in a batch of labels and returns the encoded labels in the following format:
+        List of length N containing tensors of shape (M, 5), where N is the number of samples,
+        M is the number of bounding boxes in the sample and each bounding box is represented by 5 values: (class_id,
+        x, y, w, h). x and y are the coordinates (in pixels) of the upper left corner of the bounding box, w and h are
+        the width and height of the bounding box (in pixels) and class_id is the class id.
```

Comment:
This explanation is quite long. Maybe refer to the format description above, and instead of the lengthy explanation, add an example of a transformed label?

Reply:
I think that we must state this in details, but an example is a good idea.